### PR TITLE
Create TestEnvironmentBuilder

### PR DESCRIPTION
**Pull request by sutser**

### CODE REVIEW

File: Test.Utility/TestEnvironmentBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
```
Comment (jnferner) [inline Test.Utility/TestEnvironmentBuilder.cs]: Are you sure you need all these imports?

File: Test.Utility/TestEnvironmentBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+using System.Reflection;
+using System.Text;
+using Castle.Core;
+using Messerli.Test.Utility;
+
+namespace Messerli.Test.Utility
+{
+    public class TestEnvironmentBuilder
+    {
+        private readonly List<TestFile> _testFiles = new List<TestFile>();
```
Comment (jnferner) [inline Test.Utility/TestEnvironmentBuilder.cs]: I suggest making the builder functional by using an `ImmutableList`

File: Test.Utility/TestEnvironmentBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+        public TestEnvironmentBuilder AddTestFile(TestFile file)
+        {
+            _testFiles.Add(file);
+            return this;
```
Comment (jnferner) [inline Test.Utility/TestEnvironmentBuilder.cs]: Make the builder functional by returning a new `TestEnvironmentBuilder`

File: Test.Utility.Test/TestEnvironmentBuilderTest.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        public void CreateTestEnvironmentBuilder()
+        {
+            var testEnvironmentBuilder = new TestEnvironmentBuilder();
+            
+            var testFile = new TestFile("file1.txt");
+            testEnvironmentBuilder.AddTestFile(testFile);
+            var testEnvironmentProvider = testEnvironmentBuilder.Build();
+
+            var tempPath = Path.GetTempPath();
+            Assert.True(File.Exists(Path.Combine(tempPath, testEnvironmentProvider.RootDirectory, testFile.RelativeFilePath)));
```
Comment (jnferner) [inline Test.Utility.Test/TestEnvironmentBuilderTest.cs]: Split this onto multiple lines

File: Test.Utility.Test/TestEnvironmentBuilderTest.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    public class TestEnvironmentBuilderTest 
+    {
+        [Fact]
+        public void CreateTestEnvironmentBuilder()
+        {
+            var testEnvironmentBuilder = new TestEnvironmentBuilder();
+            
+            var testFile = new TestFile("file1.txt");
+            testEnvironmentBuilder.AddTestFile(testFile);
+            var testEnvironmentProvider = testEnvironmentBuilder.Build();
```
Comment (jnferner) [inline Test.Utility.Test/TestEnvironmentBuilderTest.cs]: `new TestEnvironmentBuilder()`, `AddTestFile` and `Build` can be called fluently.

Comment (sutser) [reply Test.Utility.Test/TestEnvironmentBuilderTest.cs]: Addressed in https://github.com/messerli-informatik-ag/test-utility/pull/11/commits/c8d6ab46905a494bd91a564b0e4a71f90a2ae690

File: Test.Utility.Test/TestEnvironmentBuilderTest.cs (hunk 4 above)
Comment (sutser) [reply Test.Utility.Test/TestEnvironmentBuilderTest.cs]: Addressed in https://github.com/messerli-informatik-ag/test-utility/pull/11/commits/c8d6ab46905a494bd91a564b0e4a71f90a2ae690

File: Test.Utility/TestEnvironmentBuilder.cs (hunk 3 above)
Comment (sutser) [reply Test.Utility/TestEnvironmentBuilder.cs]: Addressed in https://github.com/messerli-informatik-ag/test-utility/pull/11/commits/c8d6ab46905a494bd91a564b0e4a71f90a2ae690

File: Test.Utility/TestEnvironmentBuilder.cs (hunk 2 above)
Comment (sutser) [reply Test.Utility/TestEnvironmentBuilder.cs]: Addressed in https://github.com/messerli-informatik-ag/test-utility/pull/11/commits/c8d6ab46905a494bd91a564b0e4a71f90a2ae690

File: Test.Utility/TestEnvironmentBuilder.cs (hunk 1 above)
Comment (sutser) [reply Test.Utility/TestEnvironmentBuilder.cs]: Addressed in https://github.com/messerli-informatik-ag/test-utility/pull/11/commits/c8d6ab46905a494bd91a564b0e4a71f90a2ae690

File: Test.Utility/TestEnvironmentBuilder.cs [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+{
+    public class TestEnvironmentBuilder
+    {
+        private readonly ImmutableList<TestFile> _testFiles = ImmutableList.Create<TestFile>();
+
+        public TestEnvironmentBuilder()
+        {
+        }
+
+        private TestEnvironmentBuilder(ImmutableList<TestFile> testFiles) => _testFiles = testFiles;
```
Comment (jnferner) [inline Test.Utility/TestEnvironmentBuilder.cs]: Split this onto multiple lines

File: paket.dependencies [hunk 7]
```diff
@@ -5,3 +5,4 @@ source https://www.nuget.org/api/v2
 nuget Autofac ~> 4.8
 nuget Moq ~> 4.10
 nuget Messerli.TempDirectory ~> 0.2
+nuget System.Collections.Immutable ~> 1.3
```
Comment (jnferner) [inline paket.dependencies]: Add a terminating newline

File: Test.Utility/TestEnvironmentBuilder.cs (hunk 6 above)
Comment (sutser) [reply Test.Utility/TestEnvironmentBuilder.cs]: Addressed in https://github.com/messerli-informatik-ag/test-utility/pull/11/commits/7bb17ca31ee65c5013624ed1951fc621ceffa3dd

File: paket.dependencies (hunk 7 above)
Comment (sutser) [reply paket.dependencies]: Addressed in https://github.com/messerli-informatik-ag/test-utility/pull/11/commits/8a0c0e56a527bfa53be42bf21ad0598a499433ab

**Pull request merged**